html_text/html_text.py

```diff
@@ -47,45 +49,71 @@ def parse_html(html):
 _has_punct_before = re.compile(r'\($').search
 
 
-def selector_to_text(sel, guess_punct_space=True):
-    """ Convert a cleaned selector to text.
-    See html_text.extract_text docstring for description of the approach and options.
+def html_to_text(tree, guess_punct_space=True, guess_page_layout=False):
+    """ Convert a cleaned html tree to text.
+        See html_text.extract_text docstring for description of the approach
+        and options.
     """
-    if guess_punct_space:
-
-        def fragments():
-            prev = None
-            for text in sel.xpath('.//text()').extract():
-                if prev is not None and (_has_trailing_whitespace(prev)
-                                         or (not _has_punct_after(text) and
-                                             not _has_punct_before(prev))):
-                    yield ' '
-                yield text
-                prev = text
 
-        return _whitespace.sub(' ', ''.join(fragments()).strip())
-
-    else:
-        fragments = (x.strip() for x in sel.xpath('.//text()').extract())
-        return _whitespace.sub(' ', ' '.join(x for x in fragments if x))
-
-
-def cleaned_selector(html):
-    """ Clean selector.
-    """
-    try:
-        tree = _cleaned_html_tree(html)
-        sel = parsel.Selector(root=tree, type='html')
-    except (lxml.etree.XMLSyntaxError,
-            lxml.etree.ParseError,
-            lxml.etree.ParserError,
-            UnicodeEncodeError):
-        # likely plain text
-        sel = parsel.Selector(html)
-    return sel
-
-
-def extract_text(html, guess_punct_space=True):
+    def add_space(text, prev):
+        return (prev is not None
+                and (not _has_trailing_whitespace(prev)
+                     and (not _has_punct_after(text)
+                     and not _has_punct_before(prev)
+                          )
+                     )
+                )
```

Comment:
I think it can be more readable as several ifs, e.g.
```py
if prev is None:
    return False
if _has_trailing_whitespace(prev):
    return False
if _has_punct_after(text) or _has_punct_before(text):
    return False
return True
```

or when converted to a condition with *or*, not *and*: 

```py
return not(
    prev is None 
    or _has_trailing_whitespace(prev) 
    or _has_punct_after(text)
    or _has_punct_before(prev)
)
```

Reply:
👍